src/search/FindReplace.js

```diff
@@ -717,10 +726,43 @@ define(function (require, exports, module) {
 
     function _launchFind() {
         var editor = EditorManager.getActiveEditor();
+
         if (editor) {
+            // Make note of the original query details
+            var state = getSearchState(editor._codeMirror),
+                qry = (findBar && state && state.parsedQuery) || "",
+                idx = state && state.matchIndex;
+
             // Create a new instance of the search bar UI
             clearSearch(editor._codeMirror);
             doSearch(editor, false);
+            _findNextIfSameSearch(qry, idx, getSearchState(editor._codeMirror));
```

Comment:
This would be better as a method that simply returns a boolean, and then the findNext should be executed within _launchFind. This will make the private function more reusable and make the code easier to follow.